### ISSUE TITLE
Bug in get_balance_allowance function

### ISSUE THREAD

**Issue by senqtus** [Polymarket/py-clob-client#83]
## Overview

`get_balance_allowance` Function has input variable `params` that can be null, but function treats this variable as not nullable. 

## Description

In `get_balance_allowance` function `if params.signature_type == -1:` will throw an exception if `params` is null, therefore it should be `if params and params.signature_type == -1:` and also test should be covering this space.

## Action items

- [ ] Tag issue (bug, enhancement, feature, etc.)
- [ ] Initial discussion/assign team member
- [ ] Move discussion to PR if necessary and when ready

**Comment by nikosbosse** [comment]
Are there any updates on this?

**Comment by corneliusroemer** [comment]
Indeed, just hit this bug - this is a clear type error. I'm not sure I agree with OP that the solution is `if params and params.signature_type == -1` - instead if no balance params are passed one might want to infer signature type as well, in which case the fix would be `if params is None or params.signature_type == -1`
